File: app/src/main/java/org/carsten/WordFinderSettingsFragment.java
package org.carsten;

import android.os.Bundle;
import androidx.annotation.Nullable;
import android.text.InputFilter;
import android.text.SpannableStringBuilder;
import android.text.Spanned;

import com.takisoft.fix.support.v7.preference.EditTextPreference;
import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class WordFinderSettingsFragment extends PreferenceFragmentCompat {
	@Override
	public void onCreatePreferencesFix(Bundle savedInstanceState, String rootKey) {
		setPreferencesFromResource(R.xml.preferences, rootKey);
		EditTextPreference timeEdit = (EditTextPreference) findPreference("countdown_time_pref");
		List<InputFilter> filters = new ArrayList<>(Arrays.asList(timeEdit.getEditText().getFilters()));
		filters.add(new InputFilter() {
			@Nullable
            @Override
			public CharSequence filter(CharSequence source, int start, int end,
                                       Spanned dest, int dstart, int dend) {
				SpannableStringBuilder builder = new SpannableStringBuilder(dest);
				builder.replace(dstart, dend, source, start, end);
				if( builder.toString().matches("[0-9]+(:(|[0-5]|[0-5][0-9]))?")) //     :?( [ ?[0-5]?[0-9]?"))
					return null;
				return "";
			}
		});
		timeEdit.getEditText().setFilters(filters.toArray(new InputFilter[0]));
	}
}

File: app/src/main/kotlin/org/carstenf/wordfinder/gui/WordFinderSettingsFragment.kt
package org.carstenf.wordfinder.gui

import android.os.Bundle
import android.text.InputFilter
import android.text.SpannableStringBuilder
import androidx.preference.PreferenceFragmentCompat
import com.takisoft.preferencex.EditTextPreference
import org.carstenf.wordfinder.R

class WordFinderSettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences, rootKey)

        val timePref: EditTextPreference? = findPreference("countdown_time_pref") // NON-NLS
        if(timePref!=null) {
            timePref.setSummary(
                String.format(
                    getString(R.string.PrefTimeLimitSummary),
                    timePref.text
                )
            )

            timePref.setOnPreferenceChangeListener{_, newValue ->
                timePref.setSummary(
                    String.format(
                        getString(R.string.PrefTimeLimitSummary),
                        newValue
                    )
                )
                true
            }

            timePref.setOnBindEditTextListener { editText ->
                val filters = editText.filters.toMutableList()
                filters.add(InputFilter { source, start, end, dest, dstart, dend ->
                    val builder = SpannableStringBuilder(dest)
                    builder.replace(dstart, dend, source, start, end)
                    if (builder.toString().matches(Regex("[0-9]+(:([0-5]|[0-5][0-9])?)?"))) {
                        null
                    } else {
                        ""
                    }
                })
                editText.filters = filters.toTypedArray()
            }
        }
    }
}